### PR TITLE
[KEP-3973]: docs for DeploymentPodReplacementPolicy

### PR DESCRIPTION
**Pull request by atiratree**
add docs for DeploymentPodReplacementPolicy feature

- https://github.com/kubernetes/kubernetes/pull/123430
- https://github.com/kubernetes/enhancements/issues/3973

### CODE REVIEW

**Comment by drewhagen** [comment]
Heads up that `dev-1.31` branch is created now, if you choose to reopen this for the new release.

**Comment by Princesso** [comment]
Hello @atiratree :wave: please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday July 16th 2024 18:00 PST. Thank you!

**Comment by Princesso** [comment]
Hi @atiratree, please move this PR from Draft to ready-for-review, if it is ready to be reviewed, thank you!

**Comment by Princesso** [comment]
Hi @atiratree, a gentle reminder that tomorrow is the deadline for having your Docs PR ready for review. Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before tomorrow, Tuesday, July 16th, 2024 18:00 PST.

**Comment by salaxander** [comment]
Hi @atiratree, reminder that today is the deadline for having your Docs PR ready for review. Please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before today at 18:00 PST.

**Comment by atiratree** [comment]
I am sorry for the delay, the PR is ready for review.

File: content/en/docs/concepts/workloads/controllers/deployment.md [hunk 1]
```diff
@@ -1317,6 +1330,49 @@ created Pod should be ready without any of its containers crashing, for it to be
 This defaults to 0 (the Pod will be considered available as soon as it is ready). To learn more about when
 a Pod is considered ready, see [Container Probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes).
 
+### Pod Replacement Policy
+
+{{< feature-state feature_gate_name="DeploymentPodReplacementPolicy" >}}
+
+{{< note >}}
+This feature is disabled by default. You can enable it by enabling the `DeploymentPodReplacementPolicy`
```
Comment (soltysh) [inline content/en/docs/concepts/workloads/controllers/deployment.md:1338]:
```suggestion
This feature is disabled by default. You can enable it by setting the `DeploymentPodReplacementPolicy`
```

**Review by soltysh — APPROVED** [review]
Minor nit, but mostly lgtm from sig-apps pov

File: content/en/docs/concepts/workloads/controllers/deployment.md (hunk 1 above)
Comment (atiratree) [reply content/en/docs/concepts/workloads/controllers/deployment.md]: thx, fixed

**Comment by atiratree** [comment]
the [main PR](https://github.com/kubernetes/kubernetes/pull/123430) is still not merged yet
/hold

**Comment by Princesso** [comment]
Hi @atiratree, I will close this Doc PR since the associated KEP did not meet the code freeze deadline and the exception request was not approved. Feel free to reopen, target the next milestone, when it is created, or ping me if anything changes. Thank you!

/close

**Pull request closed without merging**